fancy_prompt/fancy_prompt.py

```diff
@@ -0,0 +1,189 @@
+'''
+Enhanced prompt support for UndoDB.
+Contributors: Mark Williamson, Toby Lloyd Davies
+Copyright (C) 2019 Undo Ltd
+'''
+
+from __future__ import absolute_import, division
+import re
+import datetime
+from undodb.debugger_extensions import udb
+import gdb
+
+
+# Configuration variable: apply colour to output?  Disable for TUI mode.
+colour_enabled = False
+
+
+class PromptColour(gdb.Command):
+    '''
+    Set color on or off for enhanced UndoDB prompt.
+
+    Usage: prompt-color SETTING
+
+    Arguments:
+        SETTING - on|off to enable / disable colors.
+    '''
+
+    def __init__(self):
+        super(PromptColour, self).__init__('prompt-color', gdb.COMMAND_SUPPORT)
+
+    @staticmethod
+    def invoke(argument, from_tty):
+        global colour_enabled
+        if argument == 'on':
+            colour_enabled = True
+        elif argument == 'off':
+            colour_enabled = False
```

Comment:
Maybe raise an exception if you get something else.

Reply:
Done